File: 3150-shortest-and-lexicographically-smallest-beautiful-string/shortest-and-lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string shortestBeautifulSubstring(string s, int k) {
        int i, j;
        string ans = "";
        for (i = 0; i < s.size(); i++) {
            for (j = i; j < s.size(); j++) {
                string temp = s.substr(i, j - i + 1);
                int one = 0;
                for (auto x : temp)
                    if (x == '1')
                        one++;
                if (one == k) {
                    if ((ans.empty()) || (ans.size() > temp.size()) ||
                        (ans.size() >= temp.size() && ans > temp))
                        ans = temp;
                }
            }
        }
        return ans;
    }
};

File: 3150-shortest-and-lexicographically-smallest-beautiful-string/shortest-and-lexicographically-smallest-beautiful-string.py
class Solution:
    def shortestBeautifulSubstring(self, s: str, k: int) -> str:
        ans = ""

        for i in range(len(s)):
            for j in range(i, len(s)):
                st = s[i:j+1]
                one = 0
                for x in st:
                    if x == '1':
                        one += 1
                if one == k:
                    if len(ans) == 0:
                        ans = st
                    elif len(st) < len(ans):
                        ans = st
                    elif len(st) == len(ans) and st < ans:
                        ans = st
        return ans